### DISCUSSION TITLE
How to solve uBlock issue in SvelteKit apps with Umami cloud?

### DISCUSSION THREAD

**Discussion by shyakadavis** [umami-software/umami#2374] [Q&A]
Hi.

I wanted to ask for help with overcoming the uBlock issue and/or other ad-blockers. Given that, of course, it's in no violation of a user's privacy. (I just want harmless stats.)

I am using Umami Cloud, in a SvelteKit app, and placing the script snippet in my `src/app.html` like so:
```html
<!-- Umami Analytics -->
		<script
			async
			src="https://analytics.eu.umami.is/script.js"
			data-website-id="**************"
		></script>
```
Thanks for the help.

**Comment by tony-stark-eth** [comment]
Depending on your adapter you could either use vercel rewrites or just passthrough the request with a hook to umami :)

**Reply by tony-stark-eth** [reply]
https://github.com/sveltejs/kit/pull/11537

**Reply by tony-stark-eth** [reply]
Nvm that doesn't work, but you can do a redirect in your +layout.server.ts

```
export const load = async function ({ url }) {
  if (url.pathname.startsWith('/stats')) {
    throw redirect(308, `${env.UMAMI_URL}/script.js`);
  }
}
```

**Reply by pvds** [reply]
Hi @tony-stark-eth I'm looking for a similar solution but then for a SSG Svelte app using `adapterStatic`.

The (suboptimal) solution I can think of is prerendering the /stats route:

```svelte
// src/routes/stats/+page.svelte
<script>
  window.location.replace("https://cloud.umami.is/script.js");
</script>

<meta http-equiv="refresh" content="0;url=https://cloud.umami.is/script.js">
```

```js
// src/routes/stats/+page.server.js
export const prerender = true;
```

Or specifying the redirect on the production server but getting a 404 when running locally and on CI runs, and I would like to avoid external logic as much as possible.

Any thoughts?

PS. should the redirect in your example not better be a temporary redirect and a `GET` (`302`) instead on a permanent non-`GET` (`308`)?
